core/src/api/api.rs

```diff
@@ -269,7 +275,7 @@ impl GraphAPI for GraphState {
 		Ok(self
 			.shared_state_manager
 			.read()
-			.unwrap()
+			.map_err(|_| DsnpGraphError::FailedtoReadLock(SHARED_STATE_MANAGER.to_string()))?
```

Comment:
Do we want to also return error string or something here?

Reply:
I'm not sure if I understand. Would you please explain more?